### PR TITLE
Enrich unit tests and add comment for pd failover

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

Enrich unit tests and add comment for pd failover

Related changes

 - Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
None
```

### CODE REVIEW

File: pkg/manager/member/pd_member_manager.go [hunk 1]
```diff
@@ -522,7 +522,7 @@ func (pmm *pdMemberManager) pdStatefulSetIsUpgrading(set *apps.StatefulSet, tc *
 func getFailureReplicas(tc *v1alpha1.TidbCluster) int {
 	failureReplicas := 0
 	for _, failureMember := range tc.Status.PD.FailureMembers {
-		if failureMember.MemberDeleted {
+		if !failureMember.MemberDeleted {
```
Comment (cofyc) [inline pkg/manager/member/pd_member_manager.go:525]: can you update this to reuse this function?
https://github.com/pingcap/tidb-operator/blob/master/pkg/manager/member/pd_failover.go#L101-L106

**Comment by cofyc** [comment]
started a job to run ginkgo stability tests: https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb-operator-e2e-kind-ci-master-stability/detail/tidb-operator-e2e-kind-ci-master-stability/97/pipeline

File: pkg/manager/member/pd_failover.go [hunk 2]
```diff
@@ -65,6 +65,12 @@ func NewPDFailover(cli versioned.Interface,
 		recorder}
 }
 
+// Failover is used to failover broken pd member
+// If there is 3 pd member in a tidbcluster with 1 broken pd member pd-0, pdFailover will do failover in following 3 round:
+// 1. pdFailover append pd-0 into pd status's failure Member map with nonDeleteMember state.
+// 2. pdFailover will delete pd-0 member by pdapi and mark pd-0 member in failure member map as deleted state.
+// 3. pd member manager will render pd's statefulset replicas as tc.spec.pd.replicas + count of failure pd member with deleted state.
+// If the count of the failure pd member with deleted state is equal or greater than MaxFailoverCount, we wil skip failover.
```
Comment (cofyc) [inline pkg/manager/member/pd_failover.go:73]:
```suggestion
// If there are 3 PD members in a tidb cluster with 1 broken member pd-0, pdFailover will do failover in 3 rounds:
// 1. mark pd-0 as a failure Member with non-deleted state (MemberDeleted=false)
// 2. delete the failure member pd-0, and mark it deleted (MemberDeleted=true)
// 3. PD member manager will add the count of deleted failure members more replicas
// If the count of the failure PD member with the deleted state (MemberDeleted=true) is equal or greater than MaxFailoverCount, we will skip failover.
```

Comment (cofyc) [reply pkg/manager/member/pd_failover.go:73]: - fix grammar
-  add `(MemberDeleted=true/false)` to make it clear

File: pkg/manager/member/pd_member_manager_test.go [hunk 3]
```diff
@@ -1191,6 +1191,76 @@ func TestGetNewPDSetForTidbCluster(t *testing.T) {
...
+						ComponentSpec: v1alpha1.ComponentSpec{
+							Image: "pingcap/pd:nightly",
+						},
+						Replicas: 3,
+					},
+					TiDB: v1alpha1.TiDBSpec{
+						TLSClient: &v1alpha1.TiDBTLSClient{
+							Enabled: true,
+						},
+					},
```
Comment (cofyc) [inline pkg/manager/member/pd_member_manager_test.go:1212]: can this `TiDB` field be removed? we can make our tests as short as possible

File: pkg/manager/member/pd_member_manager_test.go [hunk 4]
```diff
@@ -1191,6 +1191,76 @@ func TestGetNewPDSetForTidbCluster(t *testing.T) {
...
```
Comment (cofyc) [inline pkg/manager/member/pd_member_manager_test.go:1247]: ditto

File: pkg/manager/member/pd_member_manager_test.go (hunk 3 above)
Comment (Yisaer) [reply pkg/manager/member/pd_member_manager_test.go]: updated

**Pull request merged**